Mathlib/RingTheory/Flat/FaithfullyFlat.lean

```diff
@@ -163,6 +169,175 @@ lemma iff_flat_and_lTensor_reflects_triviality :
 
 end faithful
 
+section complex
+
+private lemma range_le_ker_of_exact_rTensor [fl : FaithfullyFlat R M]
+    ⦃N1 N2 N3 : Type*⦄
+    [AddCommGroup N1] [Module R N1]
+    [AddCommGroup N2] [Module R N2]
+    [AddCommGroup N3] [Module R N3]
+    (l12 : N1 →ₗ[R] N2) (l23 : N2 →ₗ[R] N3)
+    (ex : Function.Exact (l12.rTensor M) (l23.rTensor M)) :
+    LinearMap.range l12 ≤ LinearMap.ker l23 := by
```

Comment:
Nitpick, but I would much prefer the following formatting:
```suggestion
    ⦃N₁ : Type*⦄ [AddCommGroup N₁] [Module R N₁]
    ⦃N₂ : Type*⦄ [AddCommGroup N₂] [Module R N₂]
    ⦃N₃ : Type*⦄ [AddCommGroup N₃] [Module R N₃]
    (f : N₁ →ₗ[R] N₂) (g : N₂ →ₗ[R] N₃)
    (ex : Function.Exact (f.rTensor M) (g.rTensor M)) :
    LinearMap.range f ≤ LinearMap.ker g := by
```

Reply:
In fact, it looks to me like you could use `variable` to simplify this statement and the ones below.

Reply:
Indeed